w3af/plugins/crawl/url_payment_finder.py

```diff
@@ -0,0 +1,219 @@
+             '/shop/', '/store/', '/svc/', '/servlet/', '/cgi/', '/cgi-bin/', '/cgibin/',
+    )
+
+    _files = ('pay', 'payment', 'success', 'paymentsuccess', 'paymentcomplete', 'paymentsuccessful',
+              'successful', 'paid', 'return', 'valid', 'validpay', 'validate', 'validatepayment',
+              'validatepay', 'validation', 'complete', 'completepay', 'completepayment',
+              'trxcomplete', 'transactioncomplete', 'final', 'finished',
+    )
+    _exts = ('', '.php', '.asp', '.aspx', '.jsp', '.py', 
+             '.pl', '.rb', '.cgi', '.php3',  '.php4', '.php5',
+    )
+
+    def __init__(self):
+
+        CrawlPlugin.__init__(self)
+
+        self._headers = None
+        self._first_time = True
+        self._fuzz_images = False
+        self._seen = ScalableBloomFilter()
+
+    def crawl(self, fuzzable_request):
+        """
+        Searches for new Url's using fuzzing.
+
+        :param fuzzable_request: A fuzzable_request instance that contains
+                                    (among other things) the URL to test.
+        """
+        url = fuzzable_request.get_url()
+        self._headers = Headers([('Referer', url.url_string)])
+
+        if self._first_time:
+            self._verify_head_enabled(url)
+            self._first_time = False
+
+        # First we need to delete fragments and query strings from URL.
+        url = url.uri2url()
+
+        # And we mark this one as a "do not return" URL, because the
+        # core already found it using another technique.
+        self._seen.add(url)
+
+        self._verify_head_enabled(url)
+
+        if self._head_enabled():
+            response = self._uri_opener.HEAD(url, cache=True,
+                                             headers=self._headers)
+        else:
+            response = self._uri_opener.GET(url, cache=True,
+                                            headers=self._headers)
+
+        if response.is_text_or_html():
+            mutants_chain = chain(self._mutate_path(url))
+
+            url_repeater = repeat(url)
+            args = izip(url_repeater, mutants_chain)
+
+            self.worker_pool.map_multi_args(self._do_request, args)
+
+    def _do_request(self, url, mutant):
+        """
+        Perform a simple GET to see if the result is an error or not, and then
+        run the actual fuzzing.
+        """
+        response = self._uri_opener.GET(mutant, cache=True,
+                                        headers=self._headers)
+
+        if (response.get_code() in (403, 401) or not(is_404(response))):
+
+            # Create the fuzzable request and send it to the core
+            fr = FuzzableRequest.from_http_response(response)
+            self.output_queue.put(fr)
+            
+            #
+            #   Save it to the kb (if new)!
+            #
+            if response.get_url() not in self._seen and response.get_url().get_file_name():
```

Comment:
Here you use `self._seen` for a completely different thing. Do not use "the same DB table" to store different object types.

Reply:
Still an issue

Reply:
For this one I took the url_payment as an example and I'm not sure how to modify it.